### PR TITLE
3DMol.js Visualization

### PR DESCRIPTION
**Pull request by dgasmith**
Allow molecular visualization through `3DMol.js`. In addition, adds default units to the `to_string` functions.

### CODE REVIEW

File: qcelemental/molparse/to_string.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def to_string(molrec, dtype, units='Angstrom', atom_format=None, ghost_format=No
         Overall string format. Note that it's possible to request variations
         that don't fit the dtype spec so may not be re-readable (e.g., ghost
         and mass in nucleus label with ``'xyz'``).
-        'cfour' forces nucleus label, ignoring atom_format, ghost_format
+        'cfour' forces nucleus label, ignorming atom_format, ghost_format
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: huh?

File: qcelemental/molparse/to_string.py [hunk 2]
```diff
@@ -147,8 +162,49 @@ def to_string(molrec, dtype, units='Angstrom', atom_format=None, ghost_format=No
...
+                cnt = sep
+
+            fragments.append(list(range(cnt, len(molrec["elem"]))))
+
+        for num, frag in enumerate(fragments):
+            divider = "--"
+            if num == 0:
+                divider = ""
+
+            if any(molrec["real"][at] for at in frag):
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: is this squashing pure-ghost fragments? if so, why?

File: qcelemental/molparse/to_string.py [hunk 3]
```diff
@@ -147,8 +162,49 @@ def to_string(molrec, dtype, units='Angstrom', atom_format=None, ghost_format=No
...
+                    int(molrec["fragment_charges"][num]), molrec["fragment_multiplicities"][num]))
+
+            for at in frag:
+
+                if molrec["real"][at]:
+                    atomrepr = str(molrec["elem"][at])
+                else:
+                    atomrepr = "Gh(" + molrec["elem"][at] + ")"
+                smol.append("    {0:<8s}    {1: 14.10f} {2: 14.10f} {3: 14.10f}".format(
+                    atomrepr, geom[at][0], geom[at][1], geom[at][2]))
```
Comment (loriab) [inline qcelemental/molparse/to_string.py]: I'll keep working on this. need a live example for the joins
```
        atom_format = '{elem}'
        ghost_format = 'Gh({elem})'
        umap = {'bohr': 'au', 'angstrom': 'ang'}
        atoms = _atoms_formatter(molrec, geom, atom_format, ghost_format, width, prec, 2)

       split_atoms = np.split(atoms, molrec["fragment_separators"])
     #61  '--\nmolrec['frag'.join(split_atoms)
     smol.append(f"units {umap[units.lower()]")
```

Comment (dgasmith) [reply qcelemental/molparse/to_string.py]: I might hold off on editing the Psi4 `to_string` functionality. This was a simple move to keep the current code ticking.

Comment (loriab) [reply qcelemental/molparse/to_string.py]: sorry didn't set this up right for a PR. the following works

```
--- a/qcelemental/molparse/to_string.py
+++ b/qcelemental/molparse/to_string.py
@@ -147,6 +147,27 @@ def to_string(molrec, dtype, units='Angstrom', atom_format=None, ghost_format=No
         smol = [first_line.rstrip(), name]
         smol.extend(atoms)
 
+    elif dtype == 'psi4':
+
+        atom_format = '{elem}'
+        ghost_format = 'Gh({elem})'
+        umap = {'bohr': 'bohr', 'angstrom': 'angstrom'}
+
+        atoms = _atoms_formatter(molrec, geom, atom_format, ghost_format, width, prec, 2)
+
+        smol = [f"""{int(molrec['molecular_charge'])} {molrec['molecular_multiplicity']}"""]
+        split_atoms = np.split(atoms, molrec["fragment_separators"])
+        for ifr, fr in enumerate(split_atoms):
+            smol.extend(['--', f"""{int(molrec['fragment_charges'][ifr])} {molrec['fragment_multiplicities'][ifr]}"""])
+            smol.extend(fr.tolist())
+
+        # append units and any other non-default molecule keywords
+        smol.append(f"units {umap[units.lower()]}")
+        if molrec["fix_com"]:
+            smol.append("no_com")
+        if molrec["fix_orientation"]:
+            smol.append("no_reorient")
+
     else:
         raise ValueError(f'`to_string(dtype={dtype})` unrecognized')
 
diff --git a/qcelemental/tests/test_molparse_to_string.py b/qcelemental/tests/test_molparse_to_string.py
index 13d7400..b1d5b6b 100644
--- a/qcelemental/tests/test_molparse_to_string.py
+++ b/qcelemental/tests/test_molparse_to_string.py
@@ -94,6 +94,19 @@ Co                    0.000000000000     0.000000000000     0.000000000000
 XH                    1.058354421340     0.000000000000     0.000000000000
 H                    -1.058354421340     0.000000000000     0.000000000000
 """,
+
+"ans2_psi4_au": """0 3
+--
+0 2
+Co                    0.000000000000     0.000000000000     0.000000000000
+--
+0 2
+Gh(H)                 2.000000000000     0.000000000000     0.000000000000
+H                    -2.000000000000     0.000000000000     0.000000000000
+units bohr
+no_reorient
+""",
+
 }  # yapf: disable
 
 
@@ -109,6 +122,7 @@ H                    -1.058354421340     0.000000000000     0.000000000000
     (("subject1", {'dtype': 'xyz', 'units': 'nm', 'prec': 8, 'atom_format': '{elea}{elem}{elbl}'}), "ans1c_nm"),
     (("subject2", {'dtype': 'terachem'}), "ans2_terachem_ang"),
     (("subject2", {'dtype': 'terachem', 'units': 'bohr'}), "ans2_terachem_au"),
+    (("subject2", {'dtype': 'psi4', 'units': 'bohr'}), "ans2_psi4_au"),
```

**Pull request merged**